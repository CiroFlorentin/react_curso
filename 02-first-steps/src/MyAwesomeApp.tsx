import type { CSSProperties } from 'react';

const firstName = 'Fernando';
const lastName = 'Herrera';

const favoriteGames = ['Zelda', 'Mario', 'Metroid'];
const isActive = true;

const address = {
  street: '123 Main St',
  city: 'Somewhere',
  country: 'USA',
};

export const MyAwesomeApp = () => {
  const myStyle: CSSProperties = {
    backgroundColor: isActive ? 'green' : 'red',
    borderRadius: '8px',
    padding: 10,
  };

  return (
    <>
      <h1> {firstName} </h1>
      <h3> {lastName} </h3>
      <p> {favoriteGames.join(' , ')}</p>

      <h1>{isActive ? 'Online' : 'Offline'}</h1>
      <p style={myStyle}>{JSON.stringify(address)}</p>
    </>
  );
};

// export function MyAwesomeApp() {
//   return (
//     <>
//       <h1>Fernando</h1>
//       <h3>Herrera</h3>
//     </>
//   );
// }

// ! SE PUEDE HACER DE LAS DOS FORMAS
