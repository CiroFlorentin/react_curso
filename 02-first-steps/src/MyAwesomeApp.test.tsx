import { test, describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
    test('should render firstName and lastName', () => {

        const { container } = render(<MyAwesomeApp />)

        screen.debug()

        const h1 = container.querySelector('h1')
        const h3 = container.querySelector('h3')

        expect(h1?.innerHTML).toContain('Fernando')
        expect(h3?.innerHTML).toContain('Herrera')

        // console.log(container.innerHTML)
    })

    // test('should render firstName and lastName - screen', () => {

    //     render(<MyAwesomeApp />);

    //     screen.debug();

    //     const h1 = screen.getByRole('heading', {
    //         level: 1,
    //     })

    //     console.log(h1.innerHTML);


    //     // console.log(container.innerHTML)
    // })

    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeApp />)

        expect(container).toMatchSnapshot()

    })
});