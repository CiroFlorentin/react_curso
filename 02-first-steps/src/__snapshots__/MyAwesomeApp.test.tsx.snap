// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should march snapshot 1`] = `
<div>
  <div>
    <h1>
       
      Fernando
       
    </h1>
    <h3>
       
      Herrera
       
    </h3>
    <p>
       
      4
       
    </p>
    <p
      class="favoriteGames"
    >
       
      Zelda , Mario , Metroid
    </p>
    <h1>
      Online
    </h1>
    <p
      style="background-color: green; border-radius: 8px; padding: 10px;"
    >
      {"street":"123 Main St","city":"Somewhere","country":"USA"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div>
    <h1>
       
      Fernando
       
    </h1>
    <h3>
       
      Herrera
       
    </h3>
    <p>
       
      4
       
    </p>
    <p
      class="favoriteGames"
    >
       
      Zelda , Mario , Metroid
    </p>
    <h1>
      Online
    </h1>
    <p
      style="background-color: green; border-radius: 8px; padding: 10px;"
    >
      {"street":"123 Main St","city":"Somewhere","country":"USA"}
    </p>
  </div>
</div>
`;
